File: client/src/app/_models/monthlyBudget.ts
import { Asset } from "./asset";
import { Expense } from "./expense";

export interface MonthlyBudget {
    id: number;
    income: string;
    month: number;
    year: number;
    jobField: string;
    frequency: string;
    expenseTotal: number;
    expenses: Expense[];
    assetTotal: number;
    assets: Asset[];
    budgetParentId: number;
}

File: api/Entity/MonthlyBudget.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using api.Entity;

namespace api.DTOs
{
    [Table("MonthlyBudget")]
    public class MonthlyBudget
    {
        [Required]
        public int Id { get; set; }
        [Required]
        public float Income { get; set; }
        [Required]
        public string Frequency { get; set; }
        [Required]
        public int Month { get; set; }
        [Required]
        public int Year { get; set; }
        public string JobField { get; set; }
        public float ExpenseTotal { get; set; }
        public List<Expense> Expenses { get; set; }
        public float AssetTotal { get; set; }
        public List<Asset> Assets { get; set; }
        public int BudgetParentId { get; set; }
        public Budget BudgetParent { get; set; }
    }
}